CMakeLists.txt

```diff
@@ -0,0 +1,934 @@
+#
+#	Copyright (c) 2020 Sibi Siddharthan
+#
+cmake_minimum_required(VERSION 3.15)
```

Comment:
According to the cmake docs, the lowest I can go is 3.12.4.(Need to test) This was the first release with add_compile_options() which is used throughout build script. This means I don't have a common way of creating create symlinks anymore. I think this is okay, because in CI creating hardlinks is better.

Reply:
As the " Added support for MSVC and clang on Windows" commit (whose first line should probably read more like "cmake: support MSVC and clang on Windows") seems to require CMake v3.15, could you describe in that commit's message _why_?

Reply:
Oh, since I think of this right now: the biggest difference between the support for GCC (MINGW) on Windows and the support for MSVC/clang on Windows is that the former is expected to be built and run in Git for Windows' SDK, and the latter can easily run without the full SDK, apart from the compiler it just needs a Git Bash from an end-user Git for Windows installation (which is _much_ more light-weight than the full SDK).

I think this is really a distinction worth mentioning in both of the commits, the one introducing support for GCC (MINGW) and the one introducing support for MSVC/clang.

Also, I think that we may want to auto-initialize the `vcpkg` system similar to https://github.com/git-for-windows/git/blob/v2.26.1.windows.1/config.mak.uname#L29 (i.e. we should run `compat/vcbuild/vcpkg_install.bat` unless `compat/vcbuild/vcpkg/installed/` exists).